docs/proposals/nap-waf.md

```diff
@@ -0,0 +1,753 @@
+    HTTPRoute -->|Attached to| Gateway
+
+    %% Styling
+    classDef external fill:#e1f5fe,stroke:#01579b,stroke-width:2px
+    classDef control fill:#f3e5f5,stroke:#4a148c,stroke-width:2px
+    classDef gateway fill:#66CDAA,stroke:#333,stroke-width:2px
+    classDef wafRequired fill:#ffebee,stroke:#c62828,stroke-width:3px
+    classDef app fill:#fce4ec,stroke:#880e4f,stroke-width:2px
+    classDef volume fill:#f1f8e9,stroke:#33691e,stroke-width:2px
+    classDef endpoint fill:#FFD700,stroke:#333,stroke-width:2px
+    classDef optional fill:#f0f8ff,stroke:#4169e1,stroke-width:2px,stroke-dasharray: 5 5
+    classDef cloudAuth fill:#e8f5e8,stroke:#228b22,stroke-width:2px
+
+    class SecTeam,Compiler,Store external
+    class NGFPod control
+    class NGFServiceAccount cloudAuth
+    class Gateway,HTTPRoute gateway
+    class WafEnforcer,WafConfigMgr,PolicyVol,NginxProxy,WafPolicy wafRequired
+    class Application app
+    class PolicyVol,ConfigVol volume
+    class PublicEndpoint endpoint
+    class Secret optional
+
+    %% Notes about the architecture
+    classDef note fill:#fffacd,stroke:#daa520,stroke-width:1px,stroke-dasharray: 5 5
+    Note1[📝 Note: Architecture supports multiple Gateways<br/>with selective WAF enablement per Gateway]
+    Note2[📝 Note: Secret only required for fallback authentication<br/>Native cloud auth uses NGF Service Account annotations]
+    class Note1,Note2 note
+```
+
+This architecture diagram illustrates the complete NAP v5 WAF integration within NGINX Gateway Fabric, showing the flow from policy development to traffic protection:
+
+**External Policy Management (Blue):** Security teams develop WAF policies using NAP v5 JSON schema, compile them using NAP v5 compiler tools, and publish the compiled policy bundles to accessible storage locations (S3, HTTP servers, or MinIO).
+
+**Control Plane (Purple):** The NGF Pod in the `nginx-gateway` namespace acts as the centralized control plane, watching for NginxProxy and WafPolicy resources across application namespaces, fetching compiled policies from external storage using appropriate authentication, and distributing policy configurations to NGINX Pods via secure gRPC connections.
+
+**Data Plane (Green/Red):** When WAF is enabled through NginxProxy configuration, each Gateway deploys as a multi-container NGINX Pod containing the main NGINX container with NAP module, plus the required WAF Enforcer and WAF Config Manager containers. These containers communicate through shared ephemeral volumes rather than network calls, maintaining NAP v5's architectural requirements.
+
+**Application Namespace Resources:** All user-facing resources (Gateway, HTTPRoute, WafPolicy, NginxProxy, and optional authentication Secret) reside in application namespaces for proper isolation and RBAC management. The Secret is only required when using fallback authentication methods - native cloud authentication uses annotations on the NGF service account in the nginx-gateway namespace.
```

Comment:
`native cloud authentication uses annotations on the NGF service account in the nginx-gateway namespace`

I'd make it clear here what this authentication is for.